### PR TITLE
convert-image-format@el-amine-404: new

### PR DESCRIPTION
**Pull request by el-amine-404**
I would like to publish this Applet.

### CODE REVIEW

**Comment by anaximeno** [comment]
I believe you need to adapt some stuff for translation, try to check how it is implemented on this action https://github.com/linuxmint/cinnamon-spices-actions/blob/master/verify-sha512sum%40rcalixte/files/verify-sha512sum%40rcalixte/verify-sha512sum%40rcalixte.sh

Let me know if you have questions or need help with this.

File: convert-image-format@el-amine-404/files/convert-image-format@el-amine-404/metadata.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "description": "Convert Image Format",
+  "uuid": "convert-image-format@el-amine-404",
+  "name": "Convert Image Format",
+  "author": "amine-el-houjjaji",
```
Comment (rcalixte) [inline convert-image-format@el-amine-404/files/convert-image-format@el-amine-404/metadata.json:5]:
```suggestion
  "author": "el-amine-404",
```
This should be your GitHub username, matching the UUID.

File: convert-image-format@el-amine-404/info.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "author": "amine-el-houjjaji"
```
Comment (rcalixte) [inline convert-image-format@el-amine-404/info.json:2]:
```suggestion
  "author": "el-amine-404",
```
This should be your GitHub username, matching the UUID.

**Comment by rcalixte** [comment]
This seems to overlap with #418 as well. These need to be reviewed.

**Comment by el-amine-404** [comment]
Thank you @anaximeno and @rcalixte for your reviews.

**Comment by el-amine-404** [comment]
@anaximeno The link was very helpful 🤝🏼

The following commits are an attempt to fix the translation issue
- 5075925
- 5587f2d

**Comment by rcalixte** [comment]
> Thank you @anaximeno and @rcalixte for your reviews.

Sorry about the typo in the .json.

File: convert-image-format@el-amine-404/files/convert-image-format@el-amine-404/convert-image-format.sh [hunk 3]
```diff
@@ -0,0 +1,114 @@
+#!/bin/bash
+
+_TITLE=$"Convert Image(s) Format"
+_PROMPT=$"Choose the format to convert the image(s) to:"
+_COLUMN_1=$"Select"
+_COLUMN_2=$"Format"
+_COLUMN_3=$"Description"
+_PROGRESS_TITLE=$"Converting Images"
+_PROGRESS_TEXT=$"Processing..."
```
Comment (rcalixte) [inline convert-image-format@el-amine-404/files/convert-image-format@el-amine-404/convert-image-format.sh:9]:
```suggestion
_PROGRESS_TEXT=$"Processing..."
_NOT_AN_IMAGE=$"is not an image file and will be skipped"
```

File: convert-image-format@el-amine-404/files/convert-image-format@el-amine-404/convert-image-format.sh [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+TITLE="$(/usr/bin/gettext "$_TITLE")"
+PROMPT="$(/usr/bin/gettext "$_PROMPT")"
+COLUMN_1="$(/usr/bin/gettext "$_COLUMN_1")"
+COLUMN_2="$(/usr/bin/gettext "$_COLUMN_2")"
+COLUMN_3="$(/usr/bin/gettext "$_COLUMN_3")"
+PROGRESS_TITLE="$(/usr/bin/gettext "$_PROGRESS_TITLE")"
+PROGRESS_TEXT="$(/usr/bin/gettext "$_PROGRESS_TEXT")"
```
Comment (rcalixte) [inline convert-image-format@el-amine-404/files/convert-image-format@el-amine-404/convert-image-format.sh:17]:
```suggestion
PROGRESS_TEXT="$(/usr/bin/gettext "$_PROGRESS_TEXT")"
NOT_AN_IMAGE="$(/usr/bin/gettext "$_NOT_AN_IMAGE")"
```

File: convert-image-format@el-amine-404/files/convert-image-format@el-amine-404/convert-image-format.sh [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+  COUNT=0
+  for FILE in "$@"; do
+    MIMETYPE=$(/usr/bin/file --mime-type -b "$FILE")
+    if [[ $MIMETYPE == image/* ]]; then
+      convert_image "$FILE"
+      COUNT=$((COUNT + 1))
+      echo "$((COUNT * 100 / TOTAL_FILES))"
+      echo "# Converting $FILE ($COUNT of $TOTAL_FILES)"
+    else
+      /usr/bin/zenity --warning --text="$FILE is not an image file and will be skipped."
```
Comment (rcalixte) [inline convert-image-format@el-amine-404/files/convert-image-format@el-amine-404/convert-image-format.sh:107]:
```suggestion
      /usr/bin/zenity --warning --text="$FILE $NOT_AN_IMAGE."
```

**Comment by rcalixte** [comment]
One more string to update, then the template update one more time, and then we should be good to go!

File: convert-image-format@el-amine-404/files/convert-image-format@el-amine-404/convert-image-format.sh (hunk 3 above)
Comment (el-amine-404) [reply convert-image-format@el-amine-404/files/convert-image-format@el-amine-404/convert-image-format.sh:9]: My bad, i have missed it 😅

**Review by rcalixte — APPROVED** [review]

**Pull request merged**